src/new/wallet.js

```diff
@@ -184,6 +184,183 @@ class HathorWallet extends EventEmitter {
+          } else {
+            utxoDetails.total_utxos_locked++;
+            utxoDetails.total_amount_locked += output.value;
+          }
+        }
+      }
+    }
+
+    return utxoDetails;
+  }
+
+  /**
+   * Prepare all required data to consolidate utxos.
+   *
+   * @typedef {Object} PrepareConsolidateUtxosDataResult
+   * @property {{ address: string, value: number }[]} outputs - Destiny of the consolidated utxos
+   * @property {{ hash: string, index: number }[]} inputs - Inputs for the consolidation transaction
+   * @property {{ uid: string, name: string, symbol: string }} token - HTR or custom token
+   * @property {{ address: string, amount: number, tx_id: string, locked: boolean, index: number }[]} utxos - Array of utxos that will be consolidated
+   * @property {number} total_amount - Amount to be consolidated
+   * 
+   * @param {string} destinationAddress Address of the consolidated utxos
+   * @param {UtxoOptions} options Utxo filtering options
+   *
+   * @return {PrepareConsolidateUtxosDataResult} Required data to consolidate utxos
+   *
+   */
+  prepareConsolidateUtxosData(destinationAddress, options = {}) {
+    storage.setStore(this.store);
+    const utxoDetails = this.getUtxos({ ...options, only_available_utxos: true });
+    const inputs = [];
+    const utxos = [];
+    let total_amount = 0;
+    for (let i = 0; i < utxoDetails.utxos.length; i++) {
+      if (inputs.length === transaction.getMaxInputsConstant()) {
+        // Max number of inputs reached
+        break;
+      }
+      const utxo = utxoDetails.utxos[i];
+      inputs.push({
+        hash: utxo.tx_id,
+        index: utxo.index,
+      });
+      utxos.push(utxo);
+      total_amount += utxo.amount;
+    }
+    const outputs = [{
+      address: destinationAddress,
+      value: total_amount,
+    }];
+    const token = {
+      uid: options.token || HATHOR_TOKEN_CONFIG.uid,
+      name: '',
+      symbol: ''
+    };
+
+    return { outputs, inputs, token, utxos, total_amount };
+  }
+
+  /**
+   * Consolidates many utxos into a single one for either HTR or exactly one custom token.
+   *
+   * @typedef {Object} ConsolidationResult
+   * @property {number} total_utxos_consolidated - Number of utxos consolidated
+   * @property {number} total_amount - Consolidated amount
+   * @property {number} tx_id - Consolidated transaction id
+   * @property {{ address: string, amount: number, tx_id: string, locked: boolean, index: number }[]} utxos - Array of consolidated utxos
+   *
+   * @param {string} destinationAddress Address of the consolidated utxos
+   * @param {UtxoOptions} options Utxo filtering options
+   *
+   * @return {Promise<ConsolidationResult>} Indicates that the transaction is sent or not
+   *
+   */
+  async consolidateUtxos(destinationAddress, options = {}) {
+    storage.setStore(this.store);
+    const { outputs, inputs, token, utxos, total_amount } = this.prepareConsolidateUtxosData(destinationAddress, options);
```

Comment:
Before `prepareConsolidateUtxosData` we should validate that `destinationAddress` is from the same wallet.

Reply:
Done, thanks.